octodns/manager.py

```diff
@@ -111,30 +112,76 @@ def __init__(self, config_file, max_workers=None, include_meta=False):
             '__init__: config_file=%s (octoDNS %s)', config_file, version
         )
 
+        self._configured_sub_zones = None
+
         # Read our config file
         with open(config_file, 'r') as fh:
             self.config = safe_load(fh, enforce_order=False)
 
```

Comment:
Majority of the diffs down through this section were refactoring all the configuration and setup that `__init__` was doing into helper functions so that they can more easily be tested independently. That's used a little bit in this PR, but should hopefully pay off more down the road.  I'll mark the bits that are actually new/different with comments.